users.py

```diff
@@ -137,3 +149,92 @@ def action_to_features(self, user_action):
         sample_idx = np.random.randint(0, len(class_dataset))
         user_features, _target = class_dataset[sample_idx]
         return user_features
+
+
+
+class EMGProportionalUser(BaseUser):
+    def __init__(self):
+        self.user_policy = ProportionalUserPolicy()
+
+        self.emg_min = -128
+        self.emg_max = 127
+        self.emg_buffer = []
+
+        # needs 200 sample for first window
+        window_size = 200
+        overlap = 150
+        self.stride = window_size - overlap
+        self.n_new_samples = -overlap
+
+        self.q = multiprocessing.Queue()
+        self.p = multiprocessing.Process(target=self.worker, args=(self.q,))
+        self.p.start()
+
+        self.observation_space_ = gym.spaces.Box(
+            low=-1., high=1., shape=(8,200), dtype=np.float32)
+
+    @staticmethod
+    def worker(q):
+        m = Myo(mode=emg_mode.RAW)
+        m.connect()
+        
+        def add_to_queue(emg, movement):
+            q.put(emg)
+
+        m.add_emg_handler(add_to_queue)
+        
+        def print_battery(bat):
+            print("Battery level:", bat)
+
+        m.add_battery_handler(print_battery)
+
+        # Orange logo and bar LEDs
+        m.set_leds([128, 0, 0], [128, 0, 0])
+        # Vibrate to know we connected okay
+        m.vibrate(1)
+        
+        """worker function"""
+        while True:
+            m.run()
+        print("Worker Stopped")
+
+    def read_emg_window(self):
+        try:
+            while self.n_new_samples < self.stride:
+                while not(self.q.empty()):
+                    emg = list(self.q.get())
+                    norm_emg = np.interp(emg, (self.emg_min, self.emg_max), (-1, +1))
+                    self.emg_buffer.append(norm_emg)
+                    self.n_new_samples += 1
+
+            current_window = np.array(self.emg_buffer[-200:], dtype=np.float32)
+            self.n_new_samples = 0
+            return current_window
+
+        except KeyboardInterrupt:
+            print("Quitting")
+            quit()
+
+    @property
+    def observation_space(self):
+        return self.observation_space_
+
+    def reset(self, observation, info):
+        info["original_observation"] = observation
+        user_features = self.read_emg_window()
```

Comment:
Call this "get_emg_window" to stay in line with the queue code, hopefully we will remember it implies some wait time